fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -188,6 +198,128 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
+
+            if (materializedView == null) {
+                throw new MetaNotFoundException("Materialized view " + mvName + " is not find");
+            }
+            // check materialized view state
+            if (materializedView.getState() != OlapTableState.NORMAL) {
+                throw new DdlException("Materialized view [" + materializedView.getName() + "]'s state is not NORMAL. "
+                        + "Do not allow to do ALTER ops");
+            }
+
+            //rename materialized view
+            if (newMvName != null) {
+                processRenameMaterializedView(oldMvName, newMvName, db, materializedView);
+            } else if (refreshSchemeDesc != null) {
+                //change refresh scheme
+                processChangeRefreshScheme(refreshSchemeDesc, materializedView);
+            } else {
+                throw new DdlException("Unsupported modification for materialized view");
+            }
+        } finally {
+            db.writeUnlock();
+        }
+    }
+
+    private void processChangeRefreshScheme(RefreshSchemeDesc refreshSchemeDesc, MaterializedView materializedView)
+            throws DdlException {
+        final MaterializedView.MvRefreshScheme refreshScheme = materializedView.getRefreshScheme();
+        final String refreshType = refreshSchemeDesc.getType().name();
+
+        if (refreshSchemeDesc instanceof SyncRefreshSchemeDesc) {
+            throw new DdlException("Unsupported change to SYNC refresh type");
+        } else if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
+            AsyncRefreshSchemeDesc asyncRefreshSchemeDesc = (AsyncRefreshSchemeDesc) refreshSchemeDesc;
+            final MaterializedView.AsyncRefreshContext asyncRefreshContext = refreshScheme.getAsyncRefreshContext();
+            asyncRefreshContext.setStartTime(Utils.getLongFromDateTime(asyncRefreshSchemeDesc.getStartTime()));
+            final long step = asyncRefreshSchemeDesc.getStep();
+            final TimestampArithmeticExpr.TimeUnit timeUnit = asyncRefreshSchemeDesc.getTimeUnit();
+            if (timeUnit != null && step <= 0) {
+                throw new DdlException("Unsupported negative or zero step value");
+            }
+            asyncRefreshContext.setStep(step);
+
+            asyncRefreshContext.setTimeUnit(timeUnit);
+        }
+        final RefreshType newRefreshType = RefreshType.valueOf(refreshType);
+        refreshScheme.setType(newRefreshType);
+
+        final ChangeMaterializedViewRefreshSchemeLog log = new ChangeMaterializedViewRefreshSchemeLog(materializedView);
+        GlobalStateMgr.getCurrentState().getEditLog().logMvChangeRefreshScheme(log);
+    }
+
+    private void processRenameMaterializedView(String oldMvName, String newMvName, Database db,
+                                               MaterializedView materializedView) throws DdlException {
+        if (oldMvName.equals(newMvName)) {
+            throw new DdlException("Same materialized view name");
```

Comment:
The logic that has been verified in the Analyzer does not need to be repeatedly verified